### ISSUE TITLE
blight_input: It is not possible to map one N64 button to two different keys/buttons

### ISSUE THREAD

**Issue by richard42** [mupen64plus/mupen64plus-user-issues#164]
Originally reported on Google Code with ID 164

```
Problem:
Most PC gamepads don't resemble the N64 button layout (they usually have
less face button but carry additional analog sticks, trigger buttons, ...).
blight_input allows only one gamepad button mapping for each N64 button.
This makes it hard to find a button mapping which is comfortable for all games.

Usage example:
Some games assign individual actions to the C-buttons while other games use
them as directions. 

Solution:
To create an universal mapping that can be used for many games it might be
convenient for the user to have one button in two different locations.
Attached patch modifies blight_input so that each N64 button can be mapped
to two keyboard keys, two gamepad buttons and two axis movements. 

Technical details:
The changes made break compatibility with the old format for
blight_input.conf, the patch tries to detect old config files and convert
them to the new format. The GUI is updated to show both mappings if a
second one is present but currently only the first mapping can be changed
through the GUI. The second mapping has to be set via the config file.

(This patch depends on my other patch which I posted in Issue 142. If you
think that my other patch is incorrect then you can easily modify this
patch to apply without the other one.)
```

Reported by `ling.weak` on 2008-10-16 22:00:05

<hr>
- _Attachment: [blight_input-doubleinput.patch](https://storage.googleapis.com/google-code-attachments/mupen64plus/issue-164/comment-0/blight_input-doubleinput.patch)_

**Comment by richard42** [comment]
```
This should be added.
```

Reported by `sgorman07` on 2008-10-20 13:42:07
- **Status changed**: `Accepted`

**Comment by richard42** [comment]
```
not a defect.
```

Reported by `richard42g` on 2008-12-30 18:47:48
- **Labels added**: Type-Enhancement
- **Labels removed**: Type-Defect

**Comment by richard42** [comment]
```
I don't agree with the patch and this bug is about an old plugin. I merge it to the
bug against the new plugin mupen64plus-input-sdl
```

Reported by `sven@narfation.org` on 2013-12-18 12:40:36
- **Status changed**: `Duplicate`
- **Merged into**: #571
